### DISCUSSION TITLE
Add & Remove before upload with preview

### DISCUSSION THREAD

**Discussion by Pooyahmti** [rpldy/react-uploady#615] [Q&A]
Hi. I am designing an interface which allows the user to add as much as photos they want which are aggregated in a list and sent when a certain button is clicked.
For each item in the list there should be a preview thumbnail and a remove button which will completely remove the item from list and pending queue. 
The closest example I found was this [sandbox](https://codesandbox.io/s/react-uploady-add-and-remove-before-upload-vk1rs5?fontsize=14&hidenavigation=1&theme=dark) but in this example there is no preview and also when u click delete button the item still remains in the list.

I've tried to use **useBatchAddListener** and keep the list of files in a state but the problem is that **useBatchAddListener** doesn't pass a URL for files so  I can't render the thumbnails.
Also when using UploadPreview component I have to use **getUploadPreviewForBatchItemsMethod** because of the autoUpload set to false and using state to make a list of files inside UploadPreview component doesn't work.

I'll be happy if anyone can suggest a solution.

**Comment by yoavniran** [comment]
Hi @Pooyahmti 

I thought I had published a guide about doing this, but perhaps I started thinking or even working on one but never got around to finishing.

So, something for me to do :) 
In the meantime, you can find this [storybook](https://react-uploady-storybook.netlify.app/?path=/story/ui-upload-preview--with-different-batch-items-method) example (_WithDifferentBatchItemsMethod_) and see its [code](https://github.com/rpldy/react-uploady/blob/master/packages/ui/upload-preview/UploadPreview.stories.js#L678).

This example does something very similar to what you're asking about. 

Let me know if you have questions.

**Reply by Pooyahmti** [reply]
Thank you for answering I've checked out the example, the main problem is that clearing preview doesn't clear the upload queue.
I'll be looking forward for your guide.

**Reply by yoavniran** [reply]
clearing previews does just that. 
If you wish to actually cancel the uploads, take a look at [abortItem](https://react-uploady.org/docs/api/hooks/useAbortItem/) and [clearPending](https://react-uploady.org/docs/api/context/#clearpending)
